server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -483,6 +483,26 @@ private static void parseObjectDynamic(DocumentParserContext context, ObjectMapp
                 dynamicObjectMapper = DynamicFieldsBuilder.createDynamicObjectMapper(context, currentFieldName);
                 context.addDynamicMapper(dynamicObjectMapper);
             }
+            if (mapper.subobjects() == false) {
+                if (dynamicObjectMapper instanceof NestedObjectMapper) {
+                    throw new IllegalArgumentException(
```

Comment:
If these now get wrapped in a MapperParsingException, maybe it makes more sense to just throw MapperParsingException directly here?

Reply:
+1

Reply:
oh well, that does make sense....